api/scpca_portal/models/job.py

```diff
@@ -30,20 +36,67 @@ class Meta:
 
     # Job Information Defined from AWS (via Response)
     batch_job_id = models.TextField(null=True)
-    batch_status = models.TextField(null=True)
+    batch_status = models.TextField(null=True)  # Set by a cron job
 
     # Datasets should never be deleted
     dataset = models.ForeignKey(Dataset, null=True, on_delete=models.SET_NULL, related_name="jobs")
 
     def __str__(self):
         if self.batch_job_id:
             return f"Job {self.id} - {self.batch_job_id} - {self.state}"
-
         return f"Job {self.id} - {self.state}"
 
-    def submit(self):
-        """Submit a job via boto3, and update batch_job_id and state."""
-        pass
+    @classmethod
+    def get_job(
+        cls,
+        batch_job_name: str = None,
+        batch_job_queue: str = None,
+        batch_job_definition: str = None,
+        batch_container_overrides: dict = None,
+    ):
+        """
+        Prepare a Job instance for AWS Batch without saving it to the db.
+        """
+
+        batch_job_name = batch_job_name or "DEFAULT_BATCH_JOB_NAME"
+        batch_job_queue = batch_job_queue or "DEFAULT_BATCH_JOB_QUEUE"
+        batch_job_definition = batch_job_definition or "DEFAULT_BATCH_JOB_DEFINITION"
+        batch_container_overrides = batch_container_overrides or {"command": ["DEFAULT_COMMAND"]}
+
+        return cls(
+            batch_job_name=batch_job_name,
+            batch_job_queue=batch_job_queue,
+            batch_job_definition=batch_job_definition,
+            batch_container_overrides=batch_container_overrides,
+        )
```

Comment:
I think for the factory methods we should encapsulate the logic for different types of jobs. `get_project_job`, `get_sample_job`, and then these will be replaced with `get_dataset_job` in the future.

So these constructors would capture much of the logic in `dispatch_to_batch` but allow us to break up how we handle samples and projects.